### PR TITLE
bump pdfbox to 3.0.1

### PR DESCRIPTION
**Pull request by raymcdermott**
## Description of your pull request

Upgrade the code to align with the [major changes in pdfbox v3](https://pdfbox.apache.org/3.0/migration.html)

## Why should it be considered?

This library should maintain pace with the library that it wraps.

## Pull request checklist

Before submitting the PR make sure the following things have been done
(and denote this by checking the relevant checkboxes):

- [X] The code is consistent with [Clojure style guide](https://github.com/bbatsov/clojure-style-guide#the-clojure-style-guide).
- [X] All code passes the linter (`clj-kondo --lint src`).
- [ ] You've added tests (if possible) to cover your change(s).
- [X] All tests are passing.
- [X] The commits are consistent with [the Git commit style guide](https://chris.beams.io/posts/git-commit/).
- [ ] You've updated the [changelog](../blob/master/CHANGELOG.md) (if adding/changing user-visible functionality).

Thanks!

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -4,5 +4,6 @@
   :license {:name "BSD"
             :url "https://opensource.org/license/bsd-3-clause/"}
   :dependencies [[org.clojure/clojure "1.11.1"]
-                 [org.apache.pdfbox/pdfbox "2.0.29"]
-                 [org.apache.pdfbox/pdfbox-tools "2.0.29"]])
+                 [org.apache.pdfbox/pdfbox "3.0.1"]
+                 [org.apache.pdfbox/pdfbox "3.0.1"]
```
Comment (dotemacs) [inline project.clj:8]: A duplicate line.

File: deps.edn [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {:deps {org.clojure/clojure {:mvn/version "1.11.1"},
-        org.apache.pdfbox/pdfbox {:mvn/version "2.0.29"}
-        org.apache.pdfbox/pdfbox-tools {:mvn/version "2.0.29"}}
+        org.apache.pdfbox/pdfbox {:mvn/version "3.0.1"}
+        org.apache.pdfbox/pdfbox-io {:mvn/version "3.0.1"}
```
Comment (dotemacs) [inline deps.edn:3]: Why is this being required in here and not in `project.clj`?

Or better: does `project.clj` need this library too?

**Comment by dotemacs** [comment]
I've updated the GitHub Actions for checking out a git repo, to `v4`, as mandated by GitHub.

**Comment by raymcdermott** [comment]
Thanks for the feedback. I will make the necessary fixes

File: project.clj (hunk 1 above)
Comment (raymcdermott) [reply project.clj:8]: urgh, should be `pdfbox-io`

File: deps.edn (hunk 2 above)
Comment (raymcdermott) [reply deps.edn:3]: I assumed that you have `project.clj` because you wanted to have both `lein` and `deps.edn` support.

Is that right?

Comment (raymcdermott) [reply deps.edn:3]: yes, the duplicate line in `project.clj` should have been this dependency

**Comment by raymcdermott** [comment]
I introduced `load-pdf-from-media` & `load-pdf-from-bytes` since I could not find a method, in the new API, with the same overloads as before.

**Comment by raymcdermott** [comment]
Doh, I missed this https://pdfbox.apache.org/3.0/migration.html#changes-in-pdfbox-app

**Comment by dotemacs** [comment]
This is great, thanks for tackling this. :) 

I'll probably just squash all the commits as `PDFBox upgrade to v3`, crediting you as the author and then I'll create another release.

**Comment by dotemacs** [comment]
Closing via https://github.com/dotemacs/pdfboxing/commit/70771ab58f869363d8c2996f190396eb6fdaa04d

**Comment by dotemacs** [comment]
I've pushed a new release to Clojars: https://clojars.org/pdfboxing/versions/0.1.15.4-SNAPSHOT

I'll wait until clj-kondo's linter gets fixed to cut another release.

Thanks again for doing this @raymcdermott 🙏

**Comment by dotemacs** [comment]
Just noticed that the latest version of PDFBox is 3.0.2! :) 

See: https://github.com/dotemacs/pdfboxing/actions/runs/8736143174

Not a big deal. Like I said above, I'll wait for the `clj-kondo` to be fixed, then I'll bump the dependencies also.

**Pull request closed without merging**